test/optim/test_zeroptimizer.py

```diff
@@ -0,0 +1,469 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+# pylint: disable=missing-module-docstring
+# pylint: disable=missing-class-docstring
+# pylint: disable=missing-function-docstring
+
+import os
+
+import numpy as np
+import unittest
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+from typing import List, Any
+import io
+from torch.optim import ZeROptimizer, SGD
+skip_if_no_cuda = unittest.skipIf(not torch.cuda.is_available(), reason="cuda required")
```

Comment:
Will the existing one below be sufficient? It's slightly different than this one, as it will also check whether each process can have exclusive access to one GPU.

https://github.com/pytorch/pytorch/blob/b5662ba0f0cb6d99805a44dd5b6bcda128468c83/torch/testing/_internal/common_distributed.py#L37